packages/core/src/domain/session/sessionCookieStore.ts

```diff
@@ -19,9 +19,20 @@ type Operations = {
   after?: (cookieSession: SessionState) => void
 }
 
+const bufferedOperations: Operations[] = []
+let onGoingOperations: Operations | undefined
```

Comment:
Since the ongoing operations can be postponed, we need to know if the current operations are the ongoing ones or not, in order to either attempt the operations or buffered them:
```ts
if (operations !== ongoingOperations) {
  bufferedOperations.push(operations)
  return
}
```

Reply:
I see. Having two functions would do the trick then, something like (the `hasOngoingOperations` flag would be true even when waiting for a retry):

```ts
const bufferedOperations: Operations[] = []
let hasOngoingOperations = false

export function withCookieLockAccess(operations: Operations) {
  if (hasOngoingOperations) {
    bufferedOperations.push(operations)
    return
  }
  hasOngoingOperations = true
  runOperations(operations, 0)
}

function runOperations(operations: Operations, numberOfRetries: number) {
   // ... rest of the current `withCookieLockAccess` logic
}

function next() { // could be renamed `onOperationsEnd`
  hasOngoingOperations = false
  const nextOperations = bufferedOperations.shift()
  if (nextOperations) {
    withCookieLockAccess(nextOperations)
  }
}
```

Reply:
It should work, yes.
Does it worth to do the change though? IMO this one move the complexity to another part of the design rather than reducing it like the first suggestion.